### PR TITLE
Generate typedefs for payload fields in `Spec` and `Status`

### PR DESCRIPTION
**Pull request by RedbackThomson**
Issue #, if available: https://github.com/aws-controllers-k8s/community/issues/864

Description of changes:
When checking payload type defs, if they are used in the spec or status are included, otherwise they are still excluded. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/testdata/models/apis/ec2/0000-00-00/generator.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    - EgressOnlyInternetGateway
+    - Fleet
+    - FpgaImage
+    - Image
+    - Instance
+    - InstanceExportTask
+    - InternetGateway
+    - KeyPair
+    - LaunchTemplateVersion
+    # - LaunchTemplate
```
Comment (jaypipes) [inline pkg/testdata/models/apis/ec2/0000-00-00/generator.yaml:20]: did you mean to not ignore launch template?

**Review by jaypipes — APPROVED** [review]
:+1: nice sleuthing on this @RedbackThomson

File: pkg/testdata/models/apis/ec2/0000-00-00/generator.yaml (hunk 1 above)
Comment (RedbackThomson) [reply pkg/testdata/models/apis/ec2/0000-00-00/generator.yaml:20]: Yes, there was an existing unit test for EC2 launch templates. Figured I'd pop this generator in there, and ignore all other resources.

**Comment by haarchri** [comment]
LGTM

**Pull request merged**